plugins/blockvolume/hostvol/hostvol_manager.go

```diff
@@ -55,21 +55,35 @@ func (g *glusterVolManager) GetHostingVolumesInUse() []*volume.Volinfo {
 
 // GetOrCreateHostingVolume will returns volume details for a given volume name and having a minimum size of `minSizeLimit`.
 // If volume name is not provided then it will create a gluster volume with default size for hosting gluster block.
-func (g *glusterVolManager) GetOrCreateHostingVolume(name string, minSizeLimit uint64) (*volume.Volinfo, error) {
+func (g *GlusterVolManager) GetOrCreateHostingVolume(name string, minSizeLimit uint64, hostVolumeInfo *api.HostVolumeInfo) (*volume.Volinfo, error) {
 	var (
 		volInfo      *volume.Volinfo
 		clusterLocks = transaction.Locks{}
 	)
 
-	if err := clusterLocks.Lock(path.Join(globalLockID, name)); err != nil {
-		return nil, err
+	for {
+		err := clusterLocks.Lock(path.Join(globalLockID))
```

Comment:
if host volume name is empty, then `path.Join(globalLockID, name)` resolves in `globalLockID` only. So I don't think this change is required.

Reply:
Ah ok. @aravindavk The block host volume creation is already serialized, then how can there be stale bricks? Before creating bhv it checks for alerady existing bhv under lock. So its not root cause for stale bricks?

Reply:
parallel req for BHV creation is already taken care of. This is not the root cause for any stale bricks.